BrowserKit/Sources/Logger/SentryWrapper.swift

```diff
@@ -0,0 +1,165 @@
+        guard let dsn = bundle.object(forInfoDictionaryKey: sentryDSNKey) as? String,
+              !dsn.isEmpty else {
+            return nil
+        }
+        return dsn
+    }
+
+    // MARK: - SentryWrapper protocol
+    var crashedLastLaunch: Bool {
+        return SentrySDK.crashedLastRun
+    }
+
+    func setup(sendUsageData: Bool) {
+        guard shouldSetup, sendUsageData, let dsn = dsn else { return }
+
+        SentrySDK.start { options in
+            options.dsn = dsn
+            options.environment = self.environment.rawValue
+            options.releaseName = self.releaseName
+            options.enableFileIOTracing = false
+            options.beforeBreadcrumb = { crumb in
+                if crumb.type == "http" || crumb.category == "http" {
+                    return nil
+                }
+                return crumb
+            }
+        }
+        enabled = true
+
+        configureScope()
+        configureIdentifier()
+        setupIgnoreException()
+    }
+
+    func send(message: String,
+              category: LoggerCategory,
+              level: LoggerLevel,
+              extraEvents: [String: String]?) {
+        guard shouldSendEventFor(level) else {
+            return
+        }
+
+        let event = makeEvent(message: message,
+                              category: category,
+                              level: level,
+                              extra: extraEvents)
+        captureEvent(event: event)
+    }
+
+    // MARK: - Private
+
+    private func captureEvent(event: Event) {
+        // Capture event if Sentry is enabled and a message is available
+        guard let message = event.message?.formatted else { return }
+
+        SentrySDK.capture(message: message) { (scope) in
+            scope.setEnvironment(event.environment)
+            scope.setExtras(event.extra)
+        }
+    }
+
+    private func makeEvent(message: String,
+                           category: LoggerCategory,
+                           level: LoggerLevel,
+                           extra: [String: Any]?) -> Event {
+        let event = Event(level: level.sentryLevel)
+        event.message = SentryMessage(formatted: message)
+        event.tags = ["tag": category.rawValue]
+        if let extra = extra {
+            event.extra = extra
+        }
+        return event
+    }
+
+    /// Do not send messages to Sentry if disabled OR if we are not on beta and the severity isnt severe
+    /// This is the behaviour we want for Sentry logging
+    ///       .info .warning .fatal
+    /// Debug      n        n          n
+    /// Beta         n         y          y
+    /// Release   n         n          y
+    private func shouldSendEventFor(_ level: LoggerLevel) -> Bool {
+        let shouldSendRelease = AppConstants.BuildChannel == .release && level.isGreaterOrEqualThanLevel(.fatal)
+        let shouldSendBeta = AppConstants.BuildChannel == .beta && level.isGreaterOrEqualThanLevel(.warning)
```

Comment:
Be careful setting this to warning if we are still sending many of the logs that were there before. Even if it's only beta some of them were spammy enough to cause problems.